src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -28,6 +29,28 @@
 logger = get_logger(__name__)
 
 
+def _is_guid(guid):
+    try:
+        uuid.UUID(guid)
+        return True
+    except ValueError:
+        pass
+    return False
+
+
+# Convert tenant domain name to GUID
+# https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-protocols-oidc#fetch-the-openid-connect-metadata-document
+def _to_guid_tenant_id(tenant_id):
+
+    if not _is_guid(tenant_id):
+        import requests
+        url = 'https://login.microsoftonline.com/{}/.well-known/openid-configuration'.format(tenant_id)
+        metadata = requests.get(url).json()
```

Comment:
Please handle the case when a proxy is involved, check out the [code](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L573).
Also like I mentioned endpoint like `login.microsoftonline.com` should come from current cloud configuration, again code is [here](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/util.py#L556).
Last, can we do the conversion during `az login`?

Reply:
+1 for doing the conversion at 'az login' level if possible.

Reply:
I have moved the domain resolution logic to `az login`. @yugangw-msft, please kindly review. Thanks!